### PR TITLE
Home Descriptions History

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:* /bcgov/entity#21850

*Description of changes:*
- Expands the common table component for configuration (parse prop data and slots)
- Build out template for Home Description review
- Unit test additions

Screenshot 2024-07-03 at 2 49 10 PM

Screenshot 2024-07-03 at 4 02 48 PM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1964-o8ws350b.web.app

**Review by doug-lovett — APPROVED** [review]
Fantastic!

File: ppr-ui/src/components/common/SimpleTable.vue [hunk 1]
```diff
@@ -12,28 +12,139 @@
...
+          >
+            <v-btn
+              v-if="colIndex === 0"
+              class="toggle-expand-row-btn"
+              color="primary"
+              variant="plain"
+              :ripple="false"
+              @click="toggleRowState(rowIndex)"
+            >
+              <v-btn
```
Comment (dimak1) [inline ppr-ui/src/components/common/SimpleTable.vue]: Button within a button? Maybe v-icon only will suffice with some styling?

**Review by dimak1 — APPROVED** [review]
Looks good! Just one comment.

File: ppr-ui/src/components/common/SimpleTable.vue (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/common/SimpleTable.vue]: Maybe, usually anything is possible.
The reason:
Design calls for this button-icon. The icon itself doesn't have a filled version in mdi, so by wrapping it in a 'icon' btn gives it the desired look for design requirements. 

Screenshot 2024-07-04 at 5 40 32 AM

Comment (cameron-eyds) [reply ppr-ui/src/components/common/SimpleTable.vue]: I'll try though

Comment (dimak1) [reply ppr-ui/src/components/common/SimpleTable.vue]: Ok, I see. Then we should add tabindex=-1 to the inner button and let only the parent button be accessible.

Comment (cameron-eyds) [reply ppr-ui/src/components/common/SimpleTable.vue]: Easy to re-style, no prob. Will just change to stand alone icon

Comment (dimak1) [reply ppr-ui/src/components/common/SimpleTable.vue]: Another way, is to replace the parent button with a div with a role="button" so it is accessible.

Comment (cameron-eyds) [reply ppr-ui/src/components/common/SimpleTable.vue]: Updated, looks even better now. 
Screenshot 2024-07-04 at 6 00 13 AM

Vuetify doesn't have that role pre-defined on v-btns? That has to be done explicitly everywhere??

Comment (dimak1) [reply ppr-ui/src/components/common/SimpleTable.vue]: Nice, looks good. On v-btns it has that role. If you want to have focus outline on div or row, then we need to add the role.

Comment (cameron-eyds) [reply ppr-ui/src/components/common/SimpleTable.vue]: Gotcha, thank you!

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1964-o8ws350b.web.app

**Pull request merged**